website/src/docs/concepts.mdx

```diff
@@ -4,5 +4,119 @@ export const info = {
   index: 1
 }
 
+import { Hovernote } from "/src/components/Hovernote.tsx"
+
 # Basic concepts
 
+There are only a few concepts that are needed to understand <b>garn</b>.
+The concepts are tied to the capabilities provided by the <b>garn</b> CLI.
+
+|               | Environment  |  Package   |  Check      | Executable |  Project
+|---------------|--------------|------------|-------------|------------|---------
+| command       | enter        | build      | check       | run        | --help
+| is sandboxed  | can be       | yes        | yes         | no         | --help
+
+
+## Environment
+
+An *environment* is the context in which a program runs. That includes things
+like environment variables, current working directory, state of the filesystem,
+networking stack, etc. It's what makes running the exact same program not always
+result in the exact same thing.
+
+`garn` makes creating new environments quite easy. You can easily create an
+environment where some files (for instance, executables and packages) are
+available, and variables such as PATH are set.
+
+Importantly, environments can be combined! When you run `garn enter foo`, you are
+combining your *current environment* with the environment `foo`.
```

Comment:
Personally I don't find it very helpful to think of my current environment as an `Environment` in the `garn` sense. Because it's not really something that could ever be captured by a value of type `Environment`. And I think I would find it more helpful to have this document stick to what the `Environment` type is and allows users to do.

Reply:
I do think the current environment can be given as something of type Environment! We just can’t let it “escape” certain scopes, so it’s always as a sort of continuation rather than as a top-level value. 

This actually allows us to eg write a single function that generalizes both check and shell: it’d get passed two Environment arguments, one for the defaultEnvironment and one for the current environment. If you want what’s currently a check, you just ignore the second argument. Otherwise, you combine both environments.